src/firefly/js/ui/tap/TableColumnsConstraints.jsx

```diff
@@ -171,18 +171,16 @@ function reorganizeTableModel(tableModel, columnNames, reset) {
         e.splice(constraintsColIdx, 0, '');
     });
 
-    // set selections
-    const selectInfoCls = SelectInfo.newInstance({rowCount: data.length});
-    if (selectInfoCls.getSelectedCount() === 0) {
-        let defaultSelected = [];
-        // default selected are the principal rows
-        if (isPrincipalSet(tableModel)) {
-            defaultSelected = getColumnValues(tableModel, 'principal').reduce((sels, v, i) => {
+    // default selected are either all or the principal rows
+    const usingPrincipal= isPrincipalSet(tableModel);
+    const selectInfoCls = SelectInfo.newInstance({selectAll:!usingPrincipal, rowCount: data.length});
+    if (usingPrincipal) {
```

Comment:
FIREFLY-1365 looks good. Tested it on ops and the test build side by side to compare. NED table had nothing selected on ops (previous logic), IRSA table had only principal columns selected in both (since logic for that is still the same).